### PR TITLE
Support anonymous updates via API if allowAnonymousEdits is true

### PR DESCRIPTION
**Pull request by glpatcern**
Hello,

This PR introduces the ability to use the newly provided update API anonymously, when the system is configured to work in anonymous mode (i.e. `allowAnonymousEdits` is true).

Also logging in case of error was extended with the stack trace to match similar code in the project.

The patch was successfully tested and is currently used in a pilot deployment, where CodiMD is being integrated with an external storage and controlled via API, therefore user authentication/authorization is delegated to the external storage. I can provide further information and hopefully more PRs that have helped in our integration.

### CODE REVIEW

File: lib/note/index.js [hunk 1]
```diff
@@ -292,23 +292,25 @@ const updateNote = async (req, res) => {
         title: Note.parseNoteTitle(content),
         content: content,
         lastchangeAt: now,
-        authorship: [
+        authorship: req.isAuthenticated() ? [
           [
             req.user.id,
```
Comment (Yukaii) [inline lib/note/index.js:297]: We keep authorship for the guest user, set `userId` to `null`:

```suggestion
            req.isAuthenticated() ? req.user.id : null,
```

**Review by Yukaii — CHANGES_REQUESTED** [review]

File: lib/note/index.js (hunk 1 above)
Comment (glpatcern) [reply lib/note/index.js]: Hi, makes perfect sense - actually I had thought about updating the `authorship` somehow

**Review by Yukaii — APPROVED** [review]

**Comment by Yukaii** [comment]
> I can provide further information and hopefully more PRs that have helped in our integration.

@glpatcern Thanks for the contribution 🎉 . Could you give us more details about the integration?

**Comment by glpatcern** [comment]
> @glpatcern Thanks for the contribution 🎉 . Could you give us more details about the integration?

Sure! I'll tell you more over gitter, the overall approach was simple with no radical changes (e.g. no DB schema changes) but with some UI retouches, following a bit the model of etherpad. @mic4ael has been a great contributor ;-)

**Pull request merged**